File: lc_go/lcGo/prefix/vowelStrings.go
package main

func vowelStrings(words []string, queries [][]int) []int {
	n := len(words)

	prefix := make([]int, n+1)
	for i := range prefix {
		prefix[i] = 0
	}

	isVowel := func(c byte) bool {
		return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u'
	}
	isVowelWord := func(w string) bool {
		return isVowel(w[0]) && isVowel(w[len(w)-1])
	}
	for i, w := range words {
		if isVowelWord(w) {
			prefix[i+1] = prefix[i] + 1
		} else {
			prefix[i+1] = prefix[i]
		}
	}
	ans := make([]int, len(queries))
	for i, q := range queries {
		l, r := q[0], q[1]
		ans[i] = prefix[r+1] - prefix[l]
	}
	return ans
}

func main() {

}

File: lc_cpp/lcCpp/prefix/vowelStrings.cpp
//
// Created by 24276 on 2025/8/8.
//
#include <vector>
#include <string>
class vowel_strings {
public:
    std::vector<int> vowelStrings (std::vector<std::string>& words, std::vector<std::vector<int>>& queries) {
        int n = words.size();
        std::vector<int> prefix(n + 1, 0);
        auto isVowel = [](char c ) {
            return std::string("aeiou").find(c) != std::string::npos;
        };
        auto isVowelWord = [&](const std::string& word) {
            return isVowel(word.front()) && isVowel(word.back());
        };
        for (int i = 0; i < n; i++) {
            prefix[i + 1] = prefix[i] + (isVowelWord(words[i]) ? 1 : 0);
        }
        std::vector<int> ans;

        for (auto& q :queries) {
            ans.push_back(prefix[q[1] + 1] - prefix[q[0]]);
        }
        return ans;
    }
};

int main() {
    return 0;
}
